### DISCUSSION TITLE
"Dependency failed for Open vSwitch" when installing

### DISCUSSION THREAD

**Discussion by jingzx22** [microsoft/WSL#12287] [Q&A]
Recently I found my Ubuntu WSL always encounters the **"Dependency failed for Open vSwitch"** no matter what application I am installing, say, **Valgrind**, which **has nothing to do with "Open vSwitch"** to my knowledge.

I wonder how could I manage to fix this problem. Thank you for your help!!!

The full output of the issue would be:

$ sudo apt install valgrind

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  htop libnl-genl-3-200
Use 'sudo apt autoremove' to remove them.
Suggested packages:
  valgrind-dbg valgrind-mpi kcachegrind alleyoop valkyrie
The following NEW packages will be installed:
  valgrind
0 upgraded, 1 newly installed, 0 to remove and 5 not upgraded.
1 not fully installed or removed.
Need to get 14.1 MB of archives.
After this operation, 79.3 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 valgrind amd64 1:3.18.1-1ubuntu2 [14.1 MB]
Fetched 14.1 MB in 4s (3634 kB/s)   
debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend requires a screen at least 13 lines tall and 31 columns wide.)
debconf: falling back to frontend: Readline
Selecting previously unselected package valgrind.
(Reading database ... 66203 files and directories currently installed.)
Preparing to unpack .../valgrind_1%3a3.18.1-1ubuntu2_amd64.deb ...
Unpacking valgrind (1:3.18.1-1ubuntu2) ...
Setting up openvswitch-switch (2.17.9-0ubuntu0.22.04.1) ...
Could not execute systemctl:  at /usr/bin/deb-systemd-invoke line 142.
A dependency job for openvswitch-switch.service failed. See 'journalctl -xe' for details.
invoke-rc.d: initscript openvswitch-switch, action "start" failed.
○ openvswitch-switch.service - Open vSwitch
     Loaded: loaded (/lib/systemd/system/openvswitch-switch.service; enabled; vendor preset: enabled)
     Active: inactive (dead)

Nov 18 02:25:00 LAPTOP-LOK7TL64 systemd[1]: Dependency failed for Open vSwitch.
Nov 18 02:25:00 LAPTOP-LOK7TL64 systemd[1]: openvswitch-switch.service: Job openvswitch-switch.service/start failed with result 'dependency'.
Nov 18 02:39:31 LAPTOP-LOK7TL64 systemd[1]: Dependency failed for Open vSwitch.
Nov 18 02:39:31 LAPTOP-LOK7TL64 systemd[1]: openvswitch-switch.service: Job openvswitch-switch.service/start failed with result 'dependency'.
dpkg: error processing package openvswitch-switch (--configure):
 installed openvswitch-switch package post-installation script subprocess returned error exit status 1
Setting up valgrind (1:3.18.1-1ubuntu2) ...
Processing triggers for man-db (2.10.2-1) ...
Errors were encountered while processing:
 openvswitch-switch

**Comment by clq12310** [comment]
I wonder how，too

**Comment by Realyyc** [comment]
I solve this derectly by remove the problem pkg by
```
sudo apt-get remove openvswitch-switch
```
